### PR TITLE
Default headers support

### PR DESCRIPTION
**Pull request by Kronuz**
This adds the feature in #205 by adding the following settings:

+ defaultAccept
+ defaultAcceptCharset
+ defaultAcceptDatetime
+ defaultAcceptEncoding
+ defaultAcceptLanguage
+ defaultAccessControlRequestHeaders
+ defaultAccessControlRequestMethod
+ defaultAuthorization
+ defaultCacheControl
+ defaultContentType
+ defaultCookie
+ defaultHost
+ defaultOrigin
+ defaultUserAgent

### CODE REVIEW

**Comment by Huachao** [comment]
@Kronuz Thanks for your PR, and I am considering that if we could directly provide a setting like `defaultHeaders`. And user just needs to place all their expected headers into this setting, even custom headers can be set. What's your opinion?

**Comment by Kronuz** [comment]
That sounds also good... probably even better! Still, I’d get rid of `defaultuseragent` nonetheless, so there’s only one way to define that one.

One thing I stumbled upon while making the pull request is I had to create a new instance of the `RestClientSettings` in the `HttpRequestParser`, and i wasn’t very happy about it (as it adds new callbacks to and all)... it some other way you can think of to better access it from there?

**Comment by Huachao** [comment]
I am also considering this question that in current code, every place needs to reference the settings, they have to create a new instance of the RestClientSettings class. My suggestion is that, it seems there's no need to hold many instances of setting class, can we just leverage the singleton pattern. That means add a new public static function which is responsible for creating a setting instance internally if not exists, otherwise return the created one. And the constructor of it can be even made private.

**Comment by Kronuz** [comment]
@Huachao, I modified and added what you suggested to this pull request (defaultHeaders) 👍

File: src/httpClient.ts [hunk 1]
```diff
@@ -83,8 +83,13 @@ export class HttpClient {
         }
 
         // add default user agent if not specified
```
Comment (Huachao) [inline src/httpClient.ts]: Please update the comment as well.

**Comment by Huachao** [comment]
@Kronuz please also update your code related to settings, since I have used the singleton pattern for Settings class. Thanks for your contribution.

**Comment by Huachao** [comment]
@Kronuz Merged, Thanks!

**Comment by Kronuz** [comment]
Great! will you make a release any time soon? I can't wait to use the official extension again :)

**Comment by Kronuz** [comment]
It’s ok, thank you! 👍

**Comment by Huachao** [comment]
@Kronuz you could test the latest version 0.19.0 to experience this feature, thanks for your contribution.

**Comment by aktxyz** [comment]
Can variables be used in the settings file for defaultHeaders ??

**Pull request merged**